index.d.ts

```diff
@@ -344,6 +345,11 @@ export interface TimeoutFn {
 	(ms: number): void;
 }
 
+export interface TeardownFn {
+	/** Declare a hook that is run after the test, regardless if it passed or failed. */
+	(hook: () => PromiseLike<any> | void): void;
```

Comment:
```suggestion
	/** Declare a function to be run after the test has ended. */
	(fn: () => void): void;
```

It's not a hook, so let's just say "a function"?

`void` sort of means that the return type is ignored. So `Promise<void> | void` is moot.

Reply:
> It's not a hook, so let's just say "a function"?

Cool

> `void` sort of means that the return type is ignored. So `Promise<void> | void` is moot.

I understand the meaning of `void` (generally), but not how `Promise` qualifies as `void`. Does it have some kind of special meaning in TypeScript?

Reply:
> void is a little like the opposite of any: the absence of having any type at all.

https://www.typescriptlang.org/docs/handbook/basic-types.html#void

So `void` is a superset of `Promise<void>`.

Reply:
Hum, I think I'm gonna have to drop this. Still don't see how _Promise\<T\>_ qualifies as nothing (`void`). In other words, how can `void` be _thenable_? Or do you just rely on `await` wrapping it for you?

Reply:
> Or do you just rely on `await` wrapping it for you?

Yea. It just doesn't matter. You can `await` anything.

Reply:
Sure, but still wouldn’t be able to do `teardownCallback().then(...)`.